src/DevBetterWeb.Core/Services/NewMemberService.cs

```diff
@@ -0,0 +1,140 @@
+      _emailService = emailService;
+    }
+
+    public async Task<Invitation> CreateInvitation(string email, string stripeSubscriptionId)
+    {
+      var inviteCode = Guid.NewGuid().ToString();
+      var invitation = new Invitation(email, inviteCode, stripeSubscriptionId);
+
+      await _repository.AddAsync(invitation);
+
+      return invitation;
+    }
+
+    public async Task SendRegistrationEmail(Invitation invitation)
+    {
+      string code = invitation.InviteCode;
+      string inviteEmail = invitation.Email;
+
+      string completeRegistrationUrl = GetRegistrationUrl(code, inviteEmail);
+
+      // TODO: send Discord invite here as well
+
+      var message = "Thank you for joining DevBetter! Please click to complete your registration: " + completeRegistrationUrl + "\n";
+
+      await _emailService.SendEmailAsync(inviteEmail, "Welcome to DevBetter!", message);
+    }
+
+    public async Task<string> VerifyValidEmailAndInviteCode(string email, string inviteCode)
+    {
+      var spec = new InvitationByInviteCodeWithEmailSpec(inviteCode);
+
+      string ValidEmailAndInviteCode = "success";
+
+      try
+      {
+        var storedInviteCode = await _repository.GetAsync(spec);
+        if (storedInviteCode == null)
+        {
+          throw new InvitationNotFoundException();
+        }
+        if (!storedInviteCode.Active)
+        {
+          throw new InvitationNotActiveException();
+        }
+        if (storedInviteCode.Email == null)
+        {
+          throw new InvalidEmailException();
+        }
+      }
+      catch(Exception e)
+      {
+        ValidEmailAndInviteCode = "Invalid email or invite code: " + e.GetType().ToString();
+      }
+
+      return ValidEmailAndInviteCode;
+    }
+
+    public async Task<Member> MemberSetup(string userId, string firstName, string lastName, string inviteCode)
+    {
+      Member member = CreateNewMember(userId, firstName, lastName);
+      int memberId = member.Id;
+      await AddUserToMemberRole(userId);
+
+      var spec = new InvitationByInviteCodeWithSubscriptionIdSpec(inviteCode);
+
+      var invite = await _repository.GetAsync(spec);
+      var subscriptionId = invite.PaymentHandlerSubscriptionId;
+
+      var subscriptionStart = _paymentHandlerSubscription.GetStartDate(subscriptionId);
+      var subscriptionEnd = _paymentHandlerSubscription.GetEndDate(subscriptionId);
+
+      CreateSubscriptionForMember(memberId, subscriptionStart, subscriptionEnd);
+
+      // Member has now been created and set up from the invite used. Invite should now be deactivated
+      invite.Deactivate();
+
+      return member;
+    }
+
+
+    private Member CreateNewMember(string userId, string firstName, string lastName)
+    {
+      Member member = new Member(userId);
+      member.UpdateName(firstName, lastName);
+
+      return member;
+    }
+
+    private async Task AddUserToMemberRole(string userId)
+    {
+      var roleName = "Members";
```

Comment:
This should be a constant, which probably already exists somewhere but if not, add it.